File: weyl-core/src/main/java/io/quartic/weyl/core/live/geojson/Geometry.java
package io.quartic.weyl.core.live.geojson;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property="type")
@JsonSubTypes({
        @Type(value = Point.class, name = "Point"),
        @Type(value = LineString.class, name = "LineString"),
        @Type(value = Polygon.class, name = "Polygon")
})
public interface Geometry {
}

File: common-core/src/main/kotlin/io/quartic/common/geojson/Geometry.kt
package io.quartic.common.geojson

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonSubTypes.Type

@JsonSubTypes(
        Type(value = Point::class),
        Type(value = LineString::class),
        Type(value = Polygon::class),
        Type(value = MultiPoint::class),
        Type(value = MultiLineString::class),
        Type(value = MultiPolygon::class),
        Type(value = GeometryCollection::class)
)
interface Geometry : GeoJsonObject
